hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/HoodieSparkIndexClient.java

```diff
@@ -104,7 +106,9 @@ public void create(HoodieTableMetaClient metaClient, String userIndexName, Strin
     ValidationUtils.checkState(metaClient.getIndexMetadata().isPresent(), "Index definition is not present");
 
     LOG.info("Creating index {} of using {}", fullIndexName, indexType);
-    Option<HoodieIndexDefinition> functionalIndexDefinitionOpt = Option.ofNullable(metaClient.getIndexMetadata().get().getIndexDefinitions().get(fullIndexName));
+    HoodieIndexDefinition indexDefinition = new HoodieIndexDefinition(fullIndexName, indexType, options.getOrDefault("func", SPARK_IDENTITY),
+        new ArrayList<>(columns.keySet()), options);
+    Option<HoodieIndexDefinition> functionalIndexDefinitionOpt = Option.ofNullable(indexDefinition);
```

Comment:
we would also need to update the index definition right?

Reply:
Addressed